python/sglang/srt/managers/scheduler.py

```diff
@@ -464,122 +396,59 @@ def watchdog_thread(self):
         time.sleep(5)
         self.parent_process.send_signal(signal.SIGQUIT)
 
-    @torch.no_grad()
-    def event_loop_normal(self):
-        """A normal scheduler loop."""
-        while True:
-            recv_reqs = self.recv_requests()
-            self.process_input_requests(recv_reqs)
+    def process_batch(self):
+        if self.enable_overlap:
+            return self._process_batch_overlap()
+        else:
+            return self._process_batch_normal()
 
-            batch = self.get_next_batch_to_run()
-            self.cur_batch = batch
+    @torch.no_grad()
+    def _process_batch_normal(self):
+        batch = self.get_next_batch_to_run()
+        self.cur_batch = batch
 
-            if batch:
-                result = self.run_batch(batch)
-                self.process_batch_result(batch, result)
-            else:
-                # When the server is idle, so self-check and re-init some states
-                self.check_memory()
-                self.new_token_ratio = self.init_new_token_ratio
+        if batch:
+            result = self.run_batch(batch)
+            self.process_batch_result(batch, result)
+        else:
+            # When the server is idle, so self-check and re-init some states
+            self.check_memory()
+            self.new_token_ratio = self.init_new_token_ratio
 
-            self.last_batch = batch
+        self.last_batch = batch
 
     @torch.no_grad()
```

Comment:
In the original implementation, the `@torch.no_grad()` context manager is applied to the entire loop and is only entered and exited exactly once during the entire lifecycle; while in the new implementation the `@torch.no_grad()` context manager is entered and exited every step. 

So my question is that is there any additional overhead of repeatedly calling `@torch.no_grad()`.

Reply:
I also thought of this question when modifying it. The CI benchmark does seem to pass, but yes, I can move it to a `with` block outside the loop.